File: KubicekKocnar.Server/Models/Level.cs
using Newtonsoft.Json;

namespace KubicekKocnar.Server.Models
{
    public class Level
    {
        public uint LevelId { get; set; }

        public required string Name { get; set; }

        public string? Description { get; set; }

        public uint? NextLevelId { get; set; }

        [JsonIgnore]
        public Game? Game { get; set; } = null;
        
        public uint GameId { get; set; }

        public required DateTime Created { get; set; } = DateTime.Now;

        public ICollection<PlacedBlock> Blocks { get; set; } = new List<PlacedBlock>();

        public ICollection<Feature> Features { get; set; } = new List<Feature>();

        public ICollection<Enemy> Enemies { get; set; } = new List<Enemy>();
    }
}

File: kubicekkocnar.client/src/types/Level.ts
import PlacedBlock from "./PlacedBlock"
import Game from "./Game"
import GenericFeature from "./Feature";
import { EnemyType } from "./Enemy";

// a Level is a part of a game, which can be played and contains all blocks which are currently rendered
interface Level
{
  levelId: number,
  name: string,
  description?: string,
  nextLevel?: number,

  gameId: number,
  game?: Game; // gameReference -> needs to be set on client from gameId

  blocks: PlacedBlock[];
  features: GenericFeature[];

  created: Date;

  enemies: EnemyType[];

  corruptionSpeed: number;
}

export default Level